### PR TITLE
CD-172: XClarity (Lenovo) Power Supply Health

### PR DESCRIPTION
**Pull request by mariemm97**

### CODE REVIEW

File: examples/ssh/xclarity_power_supply_health.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        ]
+    },
+    "tags": [
+        "xclarity",
+        "lenovo",
+        "power",
+        "health",
+        "status"
+    ],
+    "sample_period": 1800,
```
Comment (kgskgs) [inline examples/ssh/xclarity_power_supply_health.json:22]: change to sample_period_s

File: examples/ssh/xclarity_power_supply_health.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "version": "1.0.0",
+    "name": "XClarity Power Supply Health",
+    "description": "Monitors the health status of power supplies on Lenovo XClarity server",
+    "category": "Software Integration",
+    "requirements": {
+        "credentials": true,
+        "sandbox_version": "1.12",
+        "others": [
+            "SSH Enabled",
+            "The keyboard_interactive option can be true or false depending on the XClarity version"
```
Comment (kgskgs) [inline examples/ssh/xclarity_power_supply_health.json:12]: provides no information

**Review by kgskgs — APPROVED** [review]

File: examples/ssh/xclarity_power_supply_health.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ */
+function validate() {
+    executeCommand(command)
+        .then(parseValidateOutput)
+        .then(D.success)
+        .catch(checkSshError);
+}
+
+function parseValidateOutput(output) {
+    if (output.trim() !== "") {
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_power_supply_health.js:70]: validate properly

File: examples/ssh/xclarity_power_supply_health.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+// Parse and insert power supply health data into the custom driver table
+function parseData(output) {
+    var lines = output.split("\n");
+    var recordIdReservedWords = ['\\?', '\\*', '\\%', 'table', 'column', 'history'];
+    var recordIdSanitisationRegex = new RegExp(recordIdReservedWords.join('|'), 'g');
+    for (var i = 2; i < lines.length-1; i++) {
+        var line = lines[i].trim();
+        var values = line.split(/\s+/);
+        var recordId = (values[0] + "-" + values[1] + values[2]).replace(recordIdSanitisationRegex, '').slice(0, 50);
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_power_supply_health.js:94]: lower case it

File: examples/ssh/xclarity_power_supply_health.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+    "logo": "lenovo-xclarity.svg",
+    "version": "1.0.0",
+    "name": "XClarity Power Supply Health",
+    "description": "Monitors the health status of power supplies on Lenovo XClarity server",
+    "category": "Software Integration",
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_power_supply_health.json:6]: hardware integration

**Pull request merged**